src/Agent/NewRelic/Agent/Core/Samplers/GcSampler.cs

```diff
@@ -222,19 +222,7 @@ private int CreatePerfCounterProxies(string processInstanceName)
 
         private void LogMessage(LogLevel level, string message, Exception ex = null)
         {
-            if (!Log.IsEnabledFor(level))
-            {
-                return;
-            }
-
-            if (ex == null)
-            {
-                Log.LogMessage(level, $"GC Performance Counters: {message}");
-            }
-            else
-            {
-                Log.LogMessage(level, $"GC Performance Counters: {message}, {ex}");
-            }
+            Log.LogMessage(level, ex, $"GC Performance Counters: {message}");
```

Comment:
The previous code only incurred the string format/interpolation cost if the log level was enabled, this version will always incur that cost.

Reply:
Refactored